cpu/sam0_common/periph/spi.c

```diff
@@ -116,27 +116,25 @@ int spi_acquire(spi_t bus, spi_cs_t cs, spi_mode_t mode, spi_clk_t clk)
      * efficiency reason we do that here, so we can do all in one single write
      * to the CTRLA register */
     const uint32_t ctrla = SERCOM_SPI_CTRLA_MODE(0x3)       /* 0x3 -> master */
-                         | SERCOM_SPI_CTRLA_DOPO(spi_config[bus].mosi_pad)
-                         | SERCOM_SPI_CTRLA_DIPO(spi_config[bus].miso_pad)
-                         | (mode << SERCOM_SPI_CTRLA_CPHA_Pos);
+                           | SERCOM_SPI_CTRLA_DOPO(spi_config[bus].mosi_pad)
+                           | SERCOM_SPI_CTRLA_DIPO(spi_config[bus].miso_pad)
+                           | (mode << SERCOM_SPI_CTRLA_CPHA_Pos);
 
     /* get exclusive access to the device */
     mutex_lock(&locks[bus]);
 
     /* power on the device */
     poweron(bus);
 
-    /* configuration did not change */
-    if (dev(bus)->BAUD.reg == baud && dev(bus)->CTRLA.reg == ctrla) {
-        return SPI_OK;
-    }
-
-    /* disable the device */
-    dev(bus)->CTRLA.reg &= ~(SERCOM_SPI_CTRLA_ENABLE);
-    while (dev(bus)->SYNCBUSY.reg & SERCOM_SPI_SYNCBUSY_ENABLE) {}
+    /* first configuration or reconfiguration after altered device usage */
+    if (dev(bus)->BAUD.reg != baud || dev(bus)->CTRLA.reg != ctrla) {
```

Comment:
This statement must be rework.
If user's set the RUNSTDBY bit in their periph_conf.h then this if will always be true even if the configuration didn't change.

Reply:
I agree that this one bit could be a desired user setting. I just tried to define a mask to ignore the bit in the condition and leave it untouched during configuration. The result was that I am unable to set the `RUNSTDBY` bit after a `spi_release`, because it seems to be blocked after the `poweroff` (not writable if peripheral clock source is disabled). Removing the `poweroff` from `spi_release` enables the user to set this bit but diverges from the behavior described in the api documentation. How should I proceed?

Reply:
I just realize that RUNSTDBY bit is not setable from periph_conf.g at the moment for the SPI driver... Forget this, this is out of the scope of the PR. We'll deal with it later.

Reply:
no changes pushed concerning the matter of this conversation